app/components/edit/EditAddress.jsx

```diff
@@ -17,79 +18,142 @@ class EditAddress extends Component {
     });
   }
 
+  handleNoLocationChange(e) {
+    this.setState((state, props) => {
+      let newAddr;
+      if (state.noLocation) {
+        newAddr = this.props.address;
+      } else {
+        newAddr = {
+          "address_1": "",
+          "address_2": "",
+          "address_3": "",
+          "address_4": "",
+          "city": "",
+          "country": "",
+          "postal_code": "",
+          "state_province": ""
+        };
+      }
+      return {noLocation: !state.noLocation, address: newAddr};
+    }, () => {
+      this.props.updateAddress(this.state.address);
+    });
+  }
+
   render() {
     const address = this.props.address || {};
+    return (
+      <AddressForm handleAddressChange={this.handleAddressChange}
+      handleNoLocationChange={this.handleNoLocationChange} 
+      noLocation={this.state.noLocation} address={address} />
+    );
+  }
+}
 
+const hasNoLocation = (address) => {
+  let someFieldExistsOrNewAddress = typeof address === "undefined" || address.address_1 != "" ||
+    address.address_2 != "" || address.address_1 != "" || address.address_3 != "" || 
+    address.address_4 != "" || address.city != "" || address.postal_code != "" ||
+    address.state_province != "" || address.country != "";
+  if (someFieldExistsOrNewAddress) {
+    return false;
+  }
+  return true;
+};
+
+const AddressForm = ({noLocation, handleNoLocationChange, handleAddressChange, address}) => {
+  if (noLocation) {
     return (
       <li key="address" className="edit--section--list--item">
         <label htmlFor="address">Address</label>
+        <input
+          type="checkbox"
+          className="input-checkbox"
+          checked={noLocation}
+          onChange={handleNoLocationChange} 
+        />
+        <div onClick={handleNoLocationChange}>No Physical Location</div>
```

Comment:
Looks like this is playing the role of a label. How about you actually make it a `<label>` tag? If you have the label tag be a parent of the `<input>`, then when you click on the label text, it'll natively check the checkbox without you having to pass in an `onClick` handler. This is what I mean:

```html
<label>
  <input ...>
  No Physical Location
</label>
```

If you don't want to nest the `<input>` inside of the `<label>`, then you can give the `<input>` an `id` and have the `<label>`'s `htmlFor` attribute point to that id (note that the way we're using htmlFor everywhere else on the page is actually wrong and doesn't do anything).